tools/clang/test/HLSLFileCheck/hlsl/classes/this_cast_to_base_class.hlsl

```diff
@@ -0,0 +1,93 @@
+// RUN: %dxc -T lib_6_6 -HV 2021 -disable-lifetime-markers -fcgl %s | FileCheck %s
```

Comment:
Lifetime markers shouldn’t impact the test. If it does, your test matches are too fragile. You should be able to use a `CHECK-DAG` to work around the `bitcast` instructions inserted with lifetime markers.

Reply:
I looked at this with @python3kgae. Because the lifetime markers generate identical instructions to the instruction we need to match there isn't a good way to match around it. Disabling lifetime markers simplifies the test, so that's probably the best approach here.